lib/index.js

```diff
@@ -6,46 +6,65 @@ const PluginError = require('plugin-error')
 
 const PLUGIN_NAME = 'gulp-rollup-each'
 
+const cache = {} // cache - (outside of export)
+
 module.exports = function (arg1, arg2, injectedRollup) {
   const rollup = (injectedRollup || defaultRollup).rollup
 
   return new class extends Transform {
-    _transform (file, encoding, cb) {
+
+    _transform(file, encoding, cb) {
+
       const input = path.relative(file.cwd, file.path)
+      const ref = path.parse(input) // Input name
 
       let inputOptions = typeof arg1 === 'function' ? arg1(file) : arg1
       inputOptions = Object.assign({}, inputOptions, {
-        input: input
+        input: input,
+        cache: inputOptions.isCache ? cache[ref.name] : false
       })
 
+      // Replace `isCache` option for object
+      // Prevent Rollup throwing error of unkown `isCache` key.
+      delete Object.assign(inputOptions, {
+        ['cache']: inputOptions['isCache']
+      })['isCache'];
```

Comment:
This code overwrites the `cache` option.
I think the following code is sufficient.
```js
delete inputOptions.isCache
```

Reply:
This overwrite returns `undefined` – We need to re-assign the key `isCache`

Reply:
In the current code, `inputOptions` has the following result.
At first,
```js
// [index.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/index.js',
  cache: undefined }
// [index.js] after delete
{ plugins: ...,
  input: 'src/js/index.js',
  cache: true }

// [second.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/second.js',
  cache: undefined }
// [second.js] after delete
{ plugins: ...,
  input: 'src/js/second.js',
  cache: true }
```
After changing index.js,
```js
// [index.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/index.js',
  cache: { modules: [ [Object], [Object] ], plugins: {} } }
// [index.js] after delete
{ plugins: ...,
  input: 'src/js/index.js',
  cache: true }

// [second.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/second.js',
  cache: { modules: [ [Object], [Object] ], plugins: {} } }
// [second.js] after delete
{ plugins: ...,
  input: 'src/js/second.js',
  cache: true }
```

---

However, change to `delete inputOptions.isCache`,
```js
// [index.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/index.js',
  cache: undefined }
// [index.js] after delete
{ plugins: ...,
  input: 'src/js/index.js',
  cache: undefined }

// [second.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/second.js',
  cache: undefined }
// [second.js] after delete
{ plugins: ...,
  input: 'src/js/second.js',
  cache: true }
```
After changing index.js,
```js
// [index.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/index.js',
  cache: { modules: [ [Object], [Object] ], plugins: {} } }
// [index.js] after delete
{ plugins: ...,
  input: 'src/js/index.js',
  cache: { modules: [ [Object], [Object] ], plugins: {} } }

// [second.js] before delete
{ isCache: true,
  plugins: ...,
  input: 'src/js/second.js',
  cache: { modules: [ [Object], [Object] ], plugins: {} } }
// [second.js] after delete
{ plugins: ...,
  input: 'src/js/second.js',
  cache: { modules: [ [Object], [Object] ], plugins: {} } }
```

Reply:
`inputOptions` is generated from `arg1` every time.

Reply:
🤦🏻– I have no idea why I didn't use `delete` in the first place. I must of been awake to long.